### PR TITLE
Upgrade ES to 7.9.1 from 7.6.2; pip-compile

### PR DESCRIPTION
**Pull request by sravfeyn**
Upgrading this to latest before envs start rolling out.

### CODE REVIEW

File: docker/hq-compose.yml
Comment (millerdev) [inline docker/hq-compose.yml:76]: Will this change be a seamless upgrade for developer envs? Like is it a drop-in replacement that is fully backward compatible with existing data?

**Review by millerdev — COMMENTED** [review]
Approved unless there are manual steps necessary for dev environments, in which case we should outline those steps first.

File: docker/hq-compose.yml
Comment (sravfeyn) [reply docker/hq-compose.yml:76]: Yes, all of the code is backwards compatible. The main reason behind the upgrade is that some features are replaced in 7.9 with new features, these old features continue to work in 7.9 but will be discontinued in v8. So, we might as well use new features. These are mainly required for ICDS but it will be good to be on the latest anyway.

**Comment by sravfeyn** [comment]
@dimagi/team-commcare-hq We are again doing a minor upgrade of Elasticserach. If you are using ES7 locally, you will need to run `./manage.py preindex_everything` once this PR is merged.

**Review by millerdev — APPROVED** [review]

**Comment by sravfeyn** [comment]
@dimagi/team-commcare-hq FYI, if you are using ES7, you will need to run `./manage.py ptop_preindex` to retain your elasticsearch data

**Pull request merged**
